timely/src/dataflow/channels/mod.rs

```diff
@@ -36,10 +36,25 @@ impl<T, D> Message<T, D> {
         Message { time, data, from, seq }
     }
 
-    /// Forms a message, and pushes contents at `pusher`.
+    /// Forms a message, and pushes contents at `pusher`. Replaces `buffer` with what the pusher
+    /// leaves in place, or a new `Vec`. Note that the returned vector is always initialized to
+    /// contain [Self::default_length] elements.
```

Comment:
Word nits, but the vec probably contains zero elements, but has capacity for the default number of elements.